### PR TITLE
Fix check notes and extrachecks items

### PR DESCRIPTION
**Pull request by nanxstats**
This PR fixes a few technical details that are likely being flagged in a first CRAN submission:

- Add one more `\donttest` to fix a remaining example running > 5s check note.
- Update copyright year in license file.
- Avoid writing to user directory in the logo generation script (see DavisVaughan/extrachecks #32).

Other quality of life improvements:

- Use BibTeX to manage bibliography for vignettes and run styler on vignettes.
- Format references in `README.md` to be consistent with vignettes.
- Reformat news titles.
- Update GitHub Actions workflows.
- Use proper copyright holder name in `DESCRIPTION`.

### CODE REVIEW

File: inst/logo/logo.R [hunk 1]
```diff
@@ -2,10 +2,11 @@
 # From hexsticker.png to logo.png
 # With thanks to: https://nanx.me/blog/post/rebranding-r-packages-with-hexagon-stickers/
 
-hexSticker::sticker(subplot="inst/logo/hexsticker.png",
-        s_x = 0.98, s_y = 0.95, s_width = 1.075, s_height = 1.075,
-        package = "", p_x = 1, p_y = 1, p_size = 8, h_size = 1.2, p_family = "pf",
-        p_color = "#7B1F58", h_fill = "#EFDBE5", h_color = "#7B1F58",
-        dpi = 640, filename = "man/figures/logo.png",
```
Comment (jdblischak) [inline inst/logo/logo.R:9]: But he does save the logo file to [`man/figures/logo.png`](https://github.com/Merck/psm3mkv/blob/main/man/figures/logo.png). Also I don't think `R CMD check` is going to look in `inst/`.

Comment (nanxstats) [reply inst/logo/logo.R:9]: Remember this? 👉 https://github.com/Merck/simtrial/pull/179

Comment (jdblischak) [reply inst/logo/logo.R:9]: Only vaguely. It seems unnecessary to check `inst/`. But I guess they are worried you will `source()` R files in `inst/`.

In that case, I recommend adding `inst/logo/` to `.Rbuildignore`. This code is only for developer use. There is no need to distribute it in the source tarball to end users. In other words, CRAN should never see this file.

Comment (nanxstats) [reply inst/logo/logo.R:9]: That makes a lot of sense! Fixed in https://github.com/Merck/psm3mkv/pull/29/commits/1b5d906782b9f255aed4f7a1c450085164f72497.

**Pull request merged**